hips/hip-9997.md

```diff
@@ -0,0 +1,197 @@
+
+```
+oci://r.myreg.io/mycharts/pepper:1.2.3
+```
+
+By placing such restrictions on registry URLs and tags, Helm users are less likely to do "strange things" with charts in registries.
+
+### 5. Cache is removed
+
+Since chart packages are small in size (<1mb), the cache is hard to justify. The cache was introduced only to provide a "Docker-like" experience. While this is neat, it does not provide the user with much value.
+
+By removing the cache, much of the existing OCI features can be cut down dramatically in size. This will make OCI features much more seamless with the existing Helm user experience.
+
+### 6. `helm chart` is removed, integrated into rest of CLI
+
+Wherever possible, the subcommands provided by the new `helm chart` command should be integrated into existing Helm CLI commands.
+
+`helm chart pull` should be baked into `helm pull`, `helm chart push` should be `helm push` (new).
+
+Commands that work with the cache will be removed: `helm chart save`, `helm chart remove` and `helm chart export`.
+
+`helm registry` will be kept as is. This manages auth against OCI registries.
+
+
+### 7. Experimental until clear messaging from OCI
+
+Considering that the rest of the items above are implemented and resolved, the OCI feature set will not be made generally available until there is clear messaging from the Open Container Initiative (OCI) regarding the way that Helm is using registries.
+
+The technical details of how Helm is using the OCI distribution spec must be validated in writing, in a document approved by OCI, clearly stating that this is the correct way to publish an arbitrary artifact. Only after this will these features be taken out of experimental mode and made generally available.
+
+## Backwards compatibility
+
+Since the existing feature set is currently experimental, there will be no promise of backwards compatibility with prior OCI support (sorry!)
+
+## Security implications
+
+Registry authentication introduces a new attack vector. Upon running `helm registry login`, these credentials are stored in an unencrypted JSON file. With that said, the underlying code is leveraging docker/cli, which has the ability to store these credentials into the system keychain instead.
+
+So, while a new attack vector has been introduced, it should not be considered any less safe than using `docker login`.
+
+Note: this is already implemented in the current OCI feature set.
+
+## How to teach this
+
+Following the implementation of all facets of this HIP, extensive documentation should be added to the Helm website on how to leverage OCI.
+
+Additionally, sites containing community charts, such as [Artifact Hub](https://artifacthub.io/), should enable providers to distribute charts over OCI, and display friendly copy-paste instructions for how to download an OCI-based chart.
+
+## Reference implementation
+
+Please see [the Helm documentation](https://helm.sh/docs/topics/registries/) for existing OCI support.
```

Comment:
In a HIP, the reference implementation section demonstrates what needs to be complete before the proposal can be marked as final.

This can be interpreted as "`helm OCI registry support has been implemented", which is partially untrue.

Would you mind listing out things that need to be implemented before this proposal can be marked as "final"?

For example:

1. We need to implement `helm push`, which does not exist
1. Helm needs to implement OCI `Getter` support so that `helm dependency` and `helm pull` can support OCI registries
1. `helm chart save`, `helm chart export`, `helm chart list`, `helm chart remove`, `helm chart pull`, and `helm chart push` need to be removed
1. The OCI Artifacts proposal needs to be approved by the OCI
1. The Helm documentation for OCI support needs to be updated

Reply:
added a list similar to this